### PR TITLE
feat: dont build push if no changes

### PR DESCRIPTION
**Pull request by smuu**
Closes https://github.com/celestiaorg/knuu/issues/7

Blocked by https://github.com/celestiaorg/knuu/pull/35

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by jrmanes — APPROVED** [review]
LGTM!

**Comment by smuu** [comment]
Need to resolve conflicts.

File: pkg/knuu/instance.go [hunk 1]
```diff
@@ -259,15 +259,22 @@ func (i *Instance) Commit() error {
...
 	// TODO: To speed up the process, the image name could be dependent on the hash of the image
-	imageName, err := i.getImageRegistry()
-	if err != nil {
-		return fmt.Errorf("error getting image registry: %w", err)
-	}
-	err = i.builderFactory.PushBuilderImage(imageName)
-	if err != nil {
-		return fmt.Errorf("error pushing image for instance '%s': %w", i.name, err)
+	if i.builderFactory.Changed() {
+		// TODO: To speed up the process, the image name could be dependent on the hash of the image
```
Comment (Bidon15) [inline pkg/knuu/instance.go:263]: Do we have the same todo mentioned twice?

**Review by Bidon15 — DISMISSED** [review]
a nit that is non blocking

**Review by jrmanes — APPROVED** [review]
LGTM! 🚀

**Review by Bidon15 — APPROVED** [review]

**Pull request merged**